### ISSUE TITLE
Citrix users fail to recognize child window after several iterations

### ISSUE THREAD

**Issue by karthikkn24** [Blazemeter/CitrixPlugin#75]
Our IT Team updated Citrix to a recent/2024 release (in accordance w/ Corporate Security Policies) and now our scripts fail to work before reaching the full Test Duration.

The users fail after several successful iterations, they fail to recognize the child windows. Logs shows that the citrix session is active and is able to see the parent window and is able to send commands.

So far we have tested with Citrix versions 24.2.0.172 and 24.3.1.97

Please find the attachment for part of the the user log.

Line 1 shows that the user is on Parent window, Iteration 6 is starting.
Line 34, 35 – does Ctrl + N, which brings us the New Order dialog.
Line 39- shows that its expecting New Order dialog to appear, though the window appeared, it failed to recognize the window name. we could confirm this from screeshots that it appeared, but Jmeter doesn't recognize the child window.
[JMeter_User_Log.txt](https://github.com/user-attachments/files/16017873/JMeter_User_Log.txt)

From this iteration onwards, it couldn’t recognize the child windows.

Please help us get this resolved so we can continue using JMeter / Citrix in a secure environment.

**Comment by 3dgiordano** [comment]
Hi @karthikkn24 
Thank you for reporting this issue when using the latest versions of Citrix Workspace.

It is not within our control if Citrix Workspace does not report or provide the same information about window information as existed before.

Unfortunately, we do not have a Citrix environment to verify if the latest versions of Citrix Workspace changed something or if something stopped working.

What version of the plugin are you using?

**Comment by karthikkn24** [comment]
We are currently using 0.7.6 plugin version since 0.7.7 has issues running in command line as well as remote distributed mode.

The issue is, it works for few hours before it fails to recognize the child windows.

We could help you troubleshoot the issue if you can provide us with any tools/patches to run and provide the logs for you.

Also, FYI that Citrix "2203 LTSR" has passed End of Maintenance, and will reach EOL in 03/23/2025. So this might take you away from a supported version of Citrix.

Appreciate your response and look forward to working with you to hopefully move forward to a supported Citrix version.

**Comment by karthikkn24** [comment]
Thanks for your response.

Do you have any other users reported or ours is the first one to report on the new Citrix version?

**Comment by mhlfst** [comment]
Hi @karthikkn24,

I just came across this thread and can confirm that we are facing the same issue. Instead of the 24xx versions we keep using [2203 LTSR CU6 Hotfix 2](https://www.citrix.com/downloads/workspace-app/legacy-workspace-app-for-windows-ltsr/workspace-app-for-windows-LTSR-CU6-HF2.html).

We've informed Citrix support that there may be a bug in their 2024 versions.

**Comment by 3dgiordano** [comment]
Thanks @mhlfst 
It's good to know which versions of Citrix Workspace have problems.
Thank you for confirming this information.

The LTSR 2402 version was released, many people on the internet are reporting that they are having problems with their installation when migrating (not related with automation, more severe, doesn't work as a expected in the update of the software infrastructure) .

Citrix released the 2402 CU1 version a month ago, I don't know if you have had the opportunity to try it @mhlfst or @karthikkn24.

https://www.citrix.com/downloads/workspace-app/workspace-app-for-windows-long-term-service-release/workspace-app-for-windows-LTSR-Latest.html

However, many people on the internet report that even with CU1 they are having problems. 
They are reporting issues and hope to have them resolved by CU2 or some hotfix on CU1.

It is recommended that you review these issues with Citrix support, it still seems that version 2402 is not stable for use.

Thanks @mhlfst for sharing what version works as a expected for you. 2203 LTSR, the EOL I think is at the end of this month.
For those who work with Citrix, it is very common to stay on stable versions rather than move to unstable versions, waiting for Citrix to fix their problems.
These are issues that should be evaluated and discussed with Citrix support.

We would greatly appreciate it if you would keep us informed if Citrix releases a version that fixes your issues and is stable enough for use. That way we can also recommend that version to others.

**Comment by mhlfst** [comment]
Hi @3dgiordano,

thanks for the hint, I just tried the 2402 CU1 (LTSR) version - unfortunately with no luck, the issue persists.

According to the [Citrix Lifecycle Milestones](https://www.citrix.com/support/product-lifecycle/workspace-app.html), the 2203 **LTSR** version's EOL is not until 23-Mar-25. Nevertheless we stay in contact with Citrix. They just asked what API we use for testing, so I sent them this github repository. Do you have any further information I should provide them?

As soon as we found a new version that works, I'll let you know.
Thanks again for your great work! 👍

**Comment by karthikkn24** [comment]
@mhlfst, @3dgiordano  - Have you got any update from Citrix? we tested with LTSR 2402 CU2 which was released on Dec 12 and still facing the same issue.

**Comment by mhlfst** [comment]
Hi @karthikkn24,

Citrix has closed our first ticket on the grounds that third-party tools are not supported. We have now been able to reproduce the issue with LoadRunner and Silk Performer, so we have opened a new ticket with Citrix and are hoping for goodwill.

Also, repeatedly opening the start menu in the Citrix desktop seems to crash the Workspace App 24.2 LTSR. I have a meeting with Citrix support about this today.

If there is any news, I will let you know.

**Comment by mhlfst** [comment]
Hi @karthikkn24,

just to give you an update: Citrix are still investigating the issues. CWA 2402 CU2 didn't work for me at all. Here's an overview:

**CWA 2402 CU1:** Users can be started but fail to recognize new windows after some time

**CWA 2402 CU2:** Starting a user causes JMeter to crash due to an access violation (msvcp140.dll). Same behavior with Load Runner. Deleting some dll files in the installation directory according to [this known issue](https://docs.citrix.com/en-us/citrix-workspace-app-for-windows/2402-ltsr/whats-new/cumulative-update-2/known-issues) didn't help.

**CAW 2409:** The Blazemeter Citrix Plugin doesn't even recognize the installed Citrix Client: “No Citrix Client Receiver is installed, install it to be able to use this plugin”. Running a user anyway leads to the same access violation as with 2402 CU2.

**Comment by mhlfst** [comment]
Hi @karthikkn24,

Citrix confirmed the issue and provided a private fix for the CWA 2402 LTSR versions (case number: 83228574). After exchanging a dll file within the ICA Client directory, the issue is resolved. The fix will be included in CWA 2402 CU4 (release date unknown).

**Comment by 3dgiordano** [comment]
Hi @mhlfst 

Thank you so much for keeping us all up to date with the latest news.
We'll be eagerly awaiting the "Citrix Workspace app for Windows 2402 LTSR CU4" release, which will fix the issue you mentioned.

I hope you incorporate this fix in the next CU, version 2409, otherwise we'll all be back in the same situation with the latest versions.

For our part, we'd like to see if it's possible to incorporate the issue of not recognizing the installed version for version 2409. That way, we can look into incorporating detection and support for new versions. Can you start a new ticket providing detailed information about the issue?

@karthikkn24 The information provided by @mhlfst may be useful to you. Perhaps you can refer that information to Citrix support and they can provide it to you, as well as more information about public availability or new versions.

If any of you are Blazemeter customers, please always start a support ticket on the Blazemeter side, referencing the issue on Github. This allows us to have another channel of communication and also opening up other possibilities.

Thank you again, @mhlfst.

We'll keep the issue open. While this isn't an issue on our end, the GitHub ticket has been a channel for communicating updates regarding Citrix's resolution of these issues.
We'll keep it open until Citrix's fix is ​​publicly available.

We appreciate everyone's continued posting of any updates.
Thank you very much.

**Comment by karthikkn24** [comment]
@mhlfst - Thanks for the update, can you please provide the change suggested by Citrix? we would like to try it out.

**Comment by mhlfst** [comment]
@karthikkn24 - Unfortunately it is a private fix, so I may not provide the modified dll file to anyone else. But you could ask Citrix support referring to case number 83228574.

**Comment by karthikkn24** [comment]
@mhlfst - Thanks, I have also reached out to Citrix.
